Build a `m2w64-gcc` backwards compat package

mgorny opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Fixes #8

mgorny committed:
Build a m2w64-gcc backwards compat package

Fixes #8

Signed-off-by: Michał Górny <mgorny@quansight.com>

mgorny committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.24.11.58.28

Other tools:
- conda-build 25.9.0
- rattler-build 0.48.1
- rattler-build-conda-compat 1.4.9

isuruf commented on recipe/activate.sh:
```diff
@@ -0,0 +1,4 @@
+echo "========================================================================"
+echo "The 'm2w64-gcc' package is obsolete and is no longer updated."
+echo "Please use 'gcc' instead."
```
```suggestion
echo "Please use 'gcc, gxx, gfortran' packages instead."
```

mgorny committed:
Mention gxx & gfortran

Suggested by @isuruf.

Signed-off-by: Michał Górny <mgorny@quansight.com>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipe/recipe.yaml**:

 * ❌ The recipe must have some tests.
 * ❌ license_file entry is missing, but is required.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18801120756](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18801120756). Examine the logs at this URL for more detail.</sub>

isuruf commented:
Need to allow m2w64-gcc output for this feedstock.

isuruf commented on recipe/recipe.yaml:
```diff
@@ -0,0 +1,27 @@
+  path: .
+
+build:
+  number: ${{ build_num }}
```
Can you skip `not win`?

mgorny replied:
Yeah, sure. Sorry, didn't notice it's Windows-only — thought it's also used as a cross-compiler.

mgorny committed:
Skip on non-win

Signed-off-by: Michał Górny <mgorny@quansight.com>

mgorny committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.24.23.37.09

Other tools:
- conda-build 25.9.0
- rattler-build 0.48.1
- rattler-build-conda-compat 1.4.9
